.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,41 @@
+name: Unit Tests
+
+# Run the test suite on pushes (incl. merges) to master and dev
+# Run the test suite when a PR is opened, pushed to, or reopened
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  unit_tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.8]
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: 'pip'
+          cache-dependency-path: 'requirements**.txt'
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+          pip install -r requirements-dev.txt
+      - name: Check Import Sort
```

Comment:
I think all of these tools have accompanying GitHub actions that manage their install.  I opted to use the command line checks because 1) we have to install the `dev` requirements anyways for them to work correctly. 2) The version of the tools themselves can be managed with `requirements-dev.in`.

Reply:
Yeah, we frequently end up shelling out here for reasons similar to the ones you mentioned; i'm fine with this approach